File: js/evaluate_reverse_polish.js
#https://leetcode.com/problems/evaluate-reverse-polish-notation/submissions/962173230/

/**
 * @param {string[]} tokens
 * @return {number}
 */
const operations = {
    "+": (x, y) => x + y,
    "-": (x, y) => x - y,
    "*": (x, y) => x * y,
    "/": (x, y) => Math.trunc(x / y),
}

var evalRPN = function(tokens) {

    var stack = []
    for (let i of tokens){
        if (!(i in operations)){
            stack.push(parseInt(i));
        }else{
            var first = stack.pop()
            var temp = operations[i](stack.pop(), first);
            stack.push(temp);
        };
    };

 return stack.pop()
};

File: leetcode/evaluate_reverse_polish.py
from math import floor

map_operators = {
"+":lambda x, y:x + y,
"-": lambda x, y: x - y,
"/": lambda x, y: int(x / y),
"*": lambda x, y: x * y,
}

class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i not in map_operators:
                stack.append(int(i))
            else:
                temp = stack.pop()
                temp = map_operators[i](stack.pop(), temp)
                stack.append(temp)

        return stack.pop()